src/effects/TiltShift2.tsx

```diff
@@ -49,22 +64,24 @@ const TiltShiftShader = {
 
 export class TiltShiftEffect extends Effect {
   constructor({
-    blendFunction = BlendFunction.ADD,
-    blur = 20.0,
-    gradient = 1100.0,
-    start = [0.5, 0.0],
-    end = [0.5, 1.0],
-    delta = [1, 1],
+    blendFunction = BlendFunction.Normal,
+    blur = 1.0, // [0, 1], can go beyond 1 for extra
+    taper = 0.5, // [0, 1], can go beyond 1 for extra
+    start = [0.5, 0.0], // [0,1] percentage x,y of screenspace
+    end = [0.5, 1.0], // [0,1] percentage x,y of screenspace
+    sampleCount = 20.0, // number of blur samples
+    direction = [1, 1] // direction of blur
   } = {}) {
     super('TiltShiftEffect', TiltShiftShader.fragmentShader, {
       blendFunction,
       attributes: EffectAttribute.CONVOLUTION,
       uniforms: new Map([
         ['blur', new Uniform(blur)],
-        ['gradient', new Uniform(gradient)],
+        ['taper', new Uniform(taper)],
```

Comment:
@chasedavis why was gradient so huge in the original code, like 11000?

Reply:
That was 1100 pixels. Now if you're on a 1920x1080 canvas, a very similar look should be taper={0.5}

the math is in line 34

Basically, the diagonal size of a 1080p frame is 2203 pixels, so half that is 1101.5 pixels.

Reply:
nice! thx for the explanation